Source/Scene/ShadowMap.js

```diff
@@ -0,0 +1,397 @@
+/*global define*/
+define([
+        '../Core/BoundingRectangle',
+        '../Core/BoxOutlineGeometry',
+        '../Core/Cartesian3',
+        '../Core/Color',
+        '../Core/ColorGeometryInstanceAttribute',
+        '../Core/defined',
+        '../Core/destroyObject',
+        '../Core/GeometryInstance',
+        '../Core/Matrix4',
+        '../Core/PixelFormat',
+        '../Core/SphereOutlineGeometry',
+        '../Renderer/ClearCommand',
+        '../Renderer/Framebuffer',
+        '../Renderer/PassState',
+        '../Renderer/PixelDatatype',
+        '../Renderer/RenderState',
+        '../Renderer/ShaderProgram',
+        '../Renderer/ShaderSource',
+        '../Renderer/Texture',
+        './Camera',
+        './CullFace',
+        './OrthographicFrustum',
+        './Pass',
+        './PerInstanceColorAppearance',
+        './Primitive'
+    ], function(
+        BoundingRectangle,
+        BoxOutlineGeometry,
+        Cartesian3,
+        Color,
+        ColorGeometryInstanceAttribute,
+        defined,
+        destroyObject,
+        GeometryInstance,
+        Matrix4,
+        PixelFormat,
+        SphereOutlineGeometry,
+        ClearCommand,
+        Framebuffer,
+        PassState,
+        PixelDatatype,
+        RenderState,
+        ShaderProgram,
+        ShaderSource,
+        Texture,
+        Camera,
+        CullFace,
+        OrthographicFrustum,
+        Pass,
+        PerInstanceColorAppearance,
+        Primitive) {
+    "use strict";
+
+    // TODO : doc
+    // TODO : handle if depth texture extension is not supported
```

Comment:
If the extension is widely supported (including mobile and IE), I am fine with just requiring it.